exampleCourse/questions/element/stringInput/question.html

```diff
@@ -1,114 +1,130 @@
+      instead of the default <code>size=35</code>.
     </p>
 
 <pl-code language="python">
 "{{params.f}}"
 </pl-code>
-</pl-question-panel>
+  </pl-question-panel>
 
-<p>
-    <pl-string-input answers-name="ans7" label="Small box = " show-help-text="false" size="5"></pl-string-input>
-</p>
+  <pl-string-input answers-name="ans7" label="Small box = " show-help-text="false" size="5"></pl-string-input>
+</pl-card>
 
-<pl-question-panel>
+<pl-card>
+  <pl-question-panel>
     <p>
-      8) For students using non-latin keyboards, it is possible that non-English characters may be introduced into the box. This can cause confusion if these characters resemble a latin character (e.g., the cyrillic character <strong>&#x0430;</strong> looks a lot like the latin character <strong>a</strong>). Similarly, students may copy from text editors and paste into the answer box, causing characters like fancy quotes to be introduced. The attribute <code>normalize-to-ascii="true"</code> converts many Unicode characters to the ASCII character that mostly resembles it.
+      For students using non-latin keyboards, it is possible that non-English characters may be introduced into the
+      box. This can cause confusion if these characters resemble a latin character (e.g., the cyrillic character
+      <strong>&#x0430;</strong> looks a lot like the latin character <strong>a</strong>). Similarly, students may copy
+      from text editors and paste into the answer box, causing characters like fancy quotes to be introduced. The
+      attribute <code>normalize-to-ascii="true"</code> converts many Unicode characters to the ASCII character that
+      mostly
+      resembles it.
     </p>
     <p>
-      The box below contains an input using a Greek Epsilon, a Cyrillic ie, and fancy quotes. Copying this text into the answer box below should result in the appropriate text (<code>"Ee"</code>) being accepted.
+      The box below contains an input using a Greek Epsilon, a Cyrillic ie, and fancy quotes. Copying this text into the
+      answer box below should result in the appropriate text (<code>"Ee"</code>) being accepted.
     </p>
+    <pl-code copy-code-button="true">{{{params.g}}}</pl-code>
+  </pl-question-panel>
+  <pl-string-input answers-name="ans8" label="Weird text = " suffix="?" normalize-to-ascii="true"></pl-string-input>
+</pl-card>
 
-<div>
-{{{params.g}}}
-</div>
-</pl-question-panel>
-
-<p>
-    <pl-string-input answers-name="ans8" label="Weird text = " suffix="?" normalize-to-ascii="true"></pl-string-input>
-</p>
+<pl-card>
+  <pl-question-panel>
+    <p>
+      The input box can also allow for multiline input, done by setting <code>multiline="true"</code>. For example,
+      this is useful for cases where students are asked for the output of a program. Note that the multiline input
+      processes newlines by treating each one as the escape characters <code>\r\n</code>, so keep this in mind when
+      setting the correct answer.
```

Comment:
Would it make sense to have `pl-string-input` replace `\r\n` with `\n` in both the correct answer and the submitted answer during parse (or before comparing the strings in grade), to avoid this problem? My guess is that this may be OS-specific, so it may be worth handling properly instead of relying on instructors to take care of (which is bound to trigger more questions than it should).

Reply:
I didn't think about this too hard, but I'm not sure what people define as the "expected" behavior in this case. I guess the use of `\r\n` to denote newlines is kinda weird but I think this depends on context.

Reply:
`\r\n` is the default on Windows. `\n` is the default on Linux. I doubt that an instructor that cares about the distinction would use a multiline `pl-string-input` to make this comparison, they would likely either expect the escaped version or something more sophisticated. It is safe to assume that, in a use-case like this, for comparison purposes, any instance of `/\r*\n/` should be considered equivalent.

Reply:
That seems reasonable (weirdly enough I actually did this on Linux, maybe the browser matters?). We can just do the replacement as you suggested, note it in the docs, and it should work fine.

Reply:
Replacement here: https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/0eef7c6d48b3dff2c7d06f952f4ea4da6dfc59a8

I decided to do the replacement in parse, even if multiline is not enabled. I think this simplifies the mental model and shouldn't break any existing code.

Reply:
Parse makes sense. You've done it for the submitted answer, do you want to do it for the correct answer as well?

Reply:
This is now done for the submitted answer (in parse) and the correct answer (in grade, before it gets used).